### DISCUSSION TITLE
An example with using Javascript library

### DISCUSSION THREAD

**Discussion by shivshankardayal** [framesurge/perseus#159] [General]
An example of Perseus with a Javascript library would be very nice. For example, using CommonMark of Rust with Highlight JS.

**Comment by arctic-hen7** [comment]
There's already a JS interop example that shows how to use JS code, and from there it's pretty much as per the `wasm-bindgen` docs.

That said, I'm happy to work on a full example of this, I agree that it would be useful!

**Reply by shivshankardayal** [reply]
Thanks. More example are always better. This helps users enforce the concepts of framework.

**Reply by shivshankardayal** [reply]
There is a reason why asked for that specific example. I am working on https://github.com/Nalanda-Labs/Kunjika which is hosted at https://kunjika.ashtavakra.org. I want to write the frontend in Perseus because it provides SSR. Right now I am using Svelte. However, I have never programmed in Webassembly so an example would be mighty helpful.

**Reply by arctic-hen7** [reply]
Sure thing! Perseus should work brilliantly for that use-case (and it *should* outperform Svelte too!), and I'm happy to add some examples to help you out. If you'd like any further support, you can hop onto [the Sycamore Discord server](https://discord.com/invite/GNqWYWNTdp), where we have a channel.

**Reply by arctic-hen7** [reply]
Until I get around to this, note that the Perseus website (`website/` in this repo) does use a JS syntax highlighting library, though its usage is fairly rudimentary. Might be helpful though.

**Reply by shivshankardayal** [reply]
Thanks for the info. I will join the you guys in Discord. :)
